lib/web_ui/dev/test_platform.dart

```diff
@@ -137,58 +142,78 @@ class BrowserPlatform extends PlatformPlugin {
     final Map<String, dynamic> requestData = json.decode(payload);
     final String filename = requestData['filename'];
     final bool write = requestData['write'];
-    final String result = await _diffScreenshot(filename, write);
+    final Map<String, dynamic> region = requestData['region'];
+    final String result = await _diffScreenshot(filename, write, region);
     return shelf.Response.ok(json.encode(result));
   }
 
-  Future<String> _diffScreenshot(String filename, bool write) async {
+  Future<String> _diffScreenshot(String filename, bool write, [ Map<String, dynamic> region ]) async {
     const String _kGoldensDirectory = 'test/golden_files';
+
+    // Bail out fast if golden doesn't exist, and user doesn't want to create it.
+    final File file = File(p.join(_kGoldensDirectory, filename));
+    if (!file.existsSync() && !write) {
+      return '''
+Golden file $filename does not exist.
+
+To automatically create this file call matchGoldenFile('$filename', write: true).
+''';
+    }
+
     final wip.ChromeConnection chromeConnection =
         wip.ChromeConnection('localhost', _kChromeDevtoolsPort);
     final wip.ChromeTab chromeTab = await chromeConnection.getTab(
         (wip.ChromeTab chromeTab) => chromeTab.url.contains('localhost'));
     final wip.WipConnection wipConnection = await chromeTab.connect();
+
+    Map<String, dynamic> captureScreenshotParameters = null;
+    if (region != null) {
+      captureScreenshotParameters = {
+        'format': 'png',
+        'clip': {
+          'x': region['x'],
+          'y': region['y'],
+          'width': region['width'],
+          'height': region['height'],
+          'scale': 1, // ?
```

Comment:
It's totally OK to leave a comment saying that you don't really know what this number should be, and simply explain your intuition about the chosen number. We even have a guideline for that: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-checking-in-comments-that-ask-questions

This way others won't be afraid to improve this code, thinking that there's something special about the number.

Reply:
I added a more specific comment (basically what I told jonah above)